### PR TITLE
Added the secrets, removed DEBUG variable

### PR DESCRIPTION
**Pull request by justinorringer**
[JIRA](https://issues.redhat.com/browse/RHCLOUD-24412)

### Changes
- Linked github and gitlab secrets from vault.
- Removed DEBUG.
- Cleaned up gitlab webhook, and it now reads the token from the header

### CODE REVIEW

File: deployment/clowdapp.yml [hunk 1]
```diff
@@ -61,8 +61,17 @@ objects:
             value: ${LOGLEVEL}
           - name: DB_IMPL
             value: ${DB_IMPL}
-          - name: DEBUG
-            value: ${DEBUG}
+
```
Comment (skarekrow) [inline deployment/clowdapp.yml:64]: Extra line

File: deployment/clowdapp.yml [hunk 2]
```diff
@@ -79,6 +88,26 @@ objects:
...
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: github-secret-token
+    labels:
+      app: platform-changelog
+  data:
+    github-secret-token: 49384aec7d3f0e771e59cbabce8366f28eacae64
+  type: Opaque
+
```
Comment (skarekrow) [inline deployment/clowdapp.yml:100]: here too

File: internal/config/config.go [hunk 3]
```diff
@@ -29,7 +29,6 @@ type Config struct {
 	GitlabWebhookSecretKey string
 	Services               map[string]Service
 	Tenants                map[string]Tenant
-	Debug                  bool
```
Comment (skarekrow) [inline internal/config/config.go:32]: I think we should keep this, we may want to utilize it in the future

File: deployment/clowdapp.yml [hunk 4]
```diff
@@ -122,5 +151,3 @@ parameters:
   value: 'true'
 - name: DB_IMPL
   value: 'impl'
-- name: DEBUG
-  value: 'false'
```
Comment (skarekrow) [inline deployment/clowdapp.yml:126]: Same as my other DEBUG comment

File: internal/config/config.go [hunk 5]
```diff
@@ -89,7 +88,6 @@ func Get() *Config {
 	options.SetDefault("GithubSecretKey", os.Getenv("GITHUB_SECRET_KEY"))
 	options.SetDefault("GitlabSecretKey", os.Getenv("GITLAB_SECRET_KEY"))
 	options.SetDefault("openapi.path", "schema/openapi.yaml")
-	options.SetDefault("Debug", os.Getenv("DEBUG") == "true" || os.Getenv("DEBUG") == "1")
```
Comment (skarekrow) [inline internal/config/config.go:92]: Same as my other DEBUG comment

File: internal/config/config.go [hunk 6]
```diff
@@ -138,10 +136,10 @@ func Get() *Config {
 		Hostname:               options.GetString("Hostname"),
 		LogLevel:               options.GetString("logLevel"),
 		GithubWebhookSecretKey: options.GetString("GithubSecretKey"),
+		GitlabWebhookSecretKey: options.GetString("GitlabSecretKey"),
 		PublicPort:             options.GetString("publicPort"),
 		MetricsPort:            options.GetString("metricsPort"),
 		MetricsPath:            options.GetString("metricsPath"),
-		Debug:                  options.GetBool("Debug"),
```
Comment (skarekrow) [inline internal/config/config.go:144]: Same as my other DEBUG comment

File: internal/endpoints/github.go [hunk 7]
```diff
@@ -24,11 +23,8 @@ func (eh *EndpointHandler) GithubWebhook(w http.ResponseWriter, r *http.Request)
 
 	services := config.Get().Services
 
-	if config.Get().Debug {
-		payload, err = ioutil.ReadAll(r.Body)
-	} else {
-		payload, err = github.ValidatePayload(r, []byte(config.Get().GithubWebhookSecretKey))
-	}
+	payload, err = github.ValidatePayload(r, []byte(config.Get().GithubWebhookSecretKey))
```
Comment (skarekrow) [inline internal/endpoints/github.go:26]: We should keep the debug logic just so we can rule out validator issues in the future

File: internal/endpoints/gitlab.go [hunk 8]
```diff
@@ -100,14 +100,17 @@ func (eh *EndpointHandler) GitlabWebhook(w http.ResponseWriter, r *http.Request)
...
 
-	if config.Get().Debug {
-		payload, err = ioutil.ReadAll(r.Body)
-	} else if (r.Header["X-gitlab-token"] != nil) && (r.Header["X-gitlab-token"][0] == config.Get().GitlabWebhookSecretKey) {
-		payload, err = ioutil.ReadAll(r.Body)
-	} else {
-		err = fmt.Errorf("invalid or missing X-Gitlab-Token")
+	token := r.Header.Get("X-Gitlab-Token")
+
+	if token == "" || token != config.Get().GitlabWebhookSecretKey {
```
Comment (skarekrow) [inline internal/endpoints/gitlab.go:105]: Same as my other DEBUG comments 😂

**Review by skarekrow — CHANGES_REQUESTED** [review]

File: deployment/clowdapp.yml [hunk 9]
```diff
@@ -79,6 +88,26 @@ objects:
...
             memory: 256Mi
 
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: github-secret-token
+    labels:
+      app: platform-changelog
+  data:
+    github-secret-token: 49384aec7d3f0e771e59cbabce8366f28eacae64
```
Comment (dehort) [inline deployment/clowdapp.yml:98]: I _think_ this needs to be base64 encoded.  Same for the other secret.

File: deployment/clowdapp.yml [hunk 10]
```diff
@@ -61,8 +61,17 @@ objects:
...
+          - name: GITHUB_WEBHOOK_SECRET_TOKEN
+            valueFrom:
+              secretKeyRef:
+                key: github-secret-token
+                name: github-secret-token
+          - name: GITLAB_WEBHOOK_SECRET_TOKEN
+            valueFrom:
+              secretKeyRef:
+                key: gitlab-secret-token
+                name: gitlab-secret-token
```
Comment (dehort) [inline deployment/clowdapp.yml:74]: Remember the way the secrets are named here, needs to match how they are named in vault.

**Review by dehort — CHANGES_REQUESTED** [review]
Looking good.   I believe the values for the secrets need to be base64 encoded.

File: internal/config/config.go (hunk 3 above)
Comment (justinorringer) [reply internal/config/config.go:32]: Added that SKIP_WEBHOOK_VALIDATION var :)

File: internal/endpoints/github.go [hunk 11]
```diff
@@ -24,11 +24,12 @@ func (eh *EndpointHandler) GithubWebhook(w http.ResponseWriter, r *http.Request)
 
 	services := config.Get().Services
 
-	if config.Get().Debug {
+	if config.Get().SkipWebhookValidation {
 		payload, err = ioutil.ReadAll(r.Body)
```
Comment (dehort) [inline internal/endpoints/github.go:28]: Can you add a log message here (like in the gitlab code) telling us / warning us that we are skipping the validation?

**Review by dehort — APPROVED** [review]
Looks good to me!

**Review by skarekrow — APPROVED** [review]

**Pull request merged**